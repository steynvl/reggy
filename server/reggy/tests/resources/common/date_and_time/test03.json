{
  "type": "Date and time",
  "information": {
    "dateSeparator": "Slash, hyphen or dot",
    "timeSeparator": "Colon or dot",
    "amPmIndicator": "AM/PM",
    "leadingZeros": "Optional leading zeros",
    "dateFormats": "d/m/y/Y"
  },
  "generalRegexInfo": {
    "startRegexMatchAt": "Anywhere",
    "endRegexMatchAt": "Anywhere",
    "regexTarget": "Java"
  },
  "generateMethod": "commonUseCases"
}
