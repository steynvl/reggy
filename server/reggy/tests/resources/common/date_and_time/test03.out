Java
Pattern regex = Pattern.compile("(?:[12][0-9]|0?[1-9]|30|[12][0-9]|0?[1-9]|3[01]|[12][0-9]|0?[1-9])[-/.](?:1[0-2]|0?[1-9])[-/.][0-9]{2}[-/.][0-9]{4}");
(?:[12][0-9]|0?[1-9]|30|[12][0-9]|0?[1-9]|3[01]|[12][0-9]|0?[1-9])[-/.](?:1[0-2]|0?[1-9])[-/.][0-9]{2}[-/.][0-9]{4}
Perl
my $regex = /(?:[12][0-9]|0?[1-9]|30|[12][0-9]|0?[1-9]|3[01]|[12][0-9]|0?[1-9])[-\/.](?:1[0-2]|0?[1-9])[-\/.][0-9]{2}[-\/.][0-9]{4}/;
(?:[12][0-9]|0?[1-9]|30|[12][0-9]|0?[1-9]|3[01]|[12][0-9]|0?[1-9])[-\/.](?:1[0-2]|0?[1-9])[-\/.][0-9]{2}[-\/.][0-9]{4}
POSIX
(?:[12][0-9]|0?[1-9]|30|[12][0-9]|0?[1-9]|3[01]|[12][0-9]|0?[1-9])[-/.](?:1[0-2]|0?[1-9])[-/.][0-9]{2}[-/.][0-9]{4}
(?:[12][0-9]|0?[1-9]|30|[12][0-9]|0?[1-9]|3[01]|[12][0-9]|0?[1-9])[-/.](?:1[0-2]|0?[1-9])[-/.][0-9]{2}[-/.][0-9]{4}
Python
regex = re.compile(r'(?:[12][0-9]|0?[1-9]|30|[12][0-9]|0?[1-9]|3[01]|[12][0-9]|0?[1-9])[-\/.](?:1[0-2]|0?[1-9])[-\/.][0-9]{2}[-\/.][0-9]{4}')
(?:[12][0-9]|0?[1-9]|30|[12][0-9]|0?[1-9]|3[01]|[12][0-9]|0?[1-9])[-\/.](?:1[0-2]|0?[1-9])[-\/.][0-9]{2}[-\/.][0-9]{4}
JavaScript
const regex = /(?:[12][0-9]|0?[1-9]|30|[12][0-9]|0?[1-9]|3[01]|[12][0-9]|0?[1-9])[-\/.](?:1[0-2]|0?[1-9])[-\/.][0-9]{2}[-\/.][0-9]{4}/;
(?:[12][0-9]|0?[1-9]|30|[12][0-9]|0?[1-9]|3[01]|[12][0-9]|0?[1-9])[-\/.](?:1[0-2]|0?[1-9])[-\/.][0-9]{2}[-\/.][0-9]{4}
PHP
$regex = '/(?:[12][0-9]|0?[1-9]|30|[12][0-9]|0?[1-9]|3[01]|[12][0-9]|0?[1-9])[-\/.](?:1[0-2]|0?[1-9])[-\/.][0-9]{2}[-\/.][0-9]{4}/';
(?:[12][0-9]|0?[1-9]|30|[12][0-9]|0?[1-9]|3[01]|[12][0-9]|0?[1-9])[-\/.](?:1[0-2]|0?[1-9])[-\/.][0-9]{2}[-\/.][0-9]{4}
Golang
regex, _ := regexp.Compile("(?:[12][0-9]|0?[1-9]|30|[12][0-9]|0?[1-9]|3[01]|[12][0-9]|0?[1-9])[-/.](?:1[0-2]|0?[1-9])[-/.][0-9]{2}[-/.][0-9]{4}")
(?:[12][0-9]|0?[1-9]|30|[12][0-9]|0?[1-9]|3[01]|[12][0-9]|0?[1-9])[-/.](?:1[0-2]|0?[1-9])[-/.][0-9]{2}[-/.][0-9]{4}
Rust
let re = Regex::new(r"(?:[12][0-9]|0?[1-9]|30|[12][0-9]|0?[1-9]|3[01]|[12][0-9]|0?[1-9])[-\/.](?:1[0-2]|0?[1-9])[-\/.][0-9]{2}[-\/.][0-9]{4}").unwrap();
(?:[12][0-9]|0?[1-9]|30|[12][0-9]|0?[1-9]|3[01]|[12][0-9]|0?[1-9])[-\/.](?:1[0-2]|0?[1-9])[-\/.][0-9]{2}[-\/.][0-9]{4}
C#
var re = new Regex(@"(?:[12][0-9]|0?[1-9]|30|[12][0-9]|0?[1-9]|3[01]|[12][0-9]|0?[1-9])[-\/.](?:1[0-2]|0?[1-9])[-\/.][0-9]{2}[-\/.][0-9]{4}");
(?:[12][0-9]|0?[1-9]|30|[12][0-9]|0?[1-9]|3[01]|[12][0-9]|0?[1-9])[-\/.](?:1[0-2]|0?[1-9])[-\/.][0-9]{2}[-\/.][0-9]{4}
Scala
val re = "(?:[12][0-9]|0?[1-9]|30|[12][0-9]|0?[1-9]|3[01]|[12][0-9]|0?[1-9])[-/.](?:1[0-2]|0?[1-9])[-/.][0-9]{2}[-/.][0-9]{4}".r
(?:[12][0-9]|0?[1-9]|30|[12][0-9]|0?[1-9]|3[01]|[12][0-9]|0?[1-9])[-/.](?:1[0-2]|0?[1-9])[-/.][0-9]{2}[-/.][0-9]{4}
Kotlin
val regex = Regex("(?:[12][0-9]|0?[1-9]|30|[12][0-9]|0?[1-9]|3[01]|[12][0-9]|0?[1-9])[-/.](?:1[0-2]|0?[1-9])[-/.][0-9]{2}[-/.][0-9]{4}")
(?:[12][0-9]|0?[1-9]|30|[12][0-9]|0?[1-9]|3[01]|[12][0-9]|0?[1-9])[-/.](?:1[0-2]|0?[1-9])[-/.][0-9]{2}[-/.][0-9]{4}